compiler/rustc_lint_defs/src/lib.rs

```diff
@@ -941,7 +950,18 @@ macro_rules! impl_lint_pass {
 #[macro_export]
 macro_rules! declare_lint_pass {
     ($(#[$m:meta])* $name:ident => [$($lint:expr),* $(,)?]) => {
-        $(#[$m])* #[derive(Copy, Clone)] pub struct $name;
+        $(#[$m])* #[derive(Copy, Clone, Default)] pub struct $name;
         $crate::impl_lint_pass!($name => [$($lint),*]);
     };
 }
+
+#[allow(rustc::lint_pass_impl_without_macro)]
+impl<P: LintPass + ?Sized> LintPass for Box<P> {
+    fn name(&self) -> &'static str {
+        (**self).name()
+    }
+
+    fn get_lints(&self) -> LintVec {
+        (**self).get_lints()
+    }
+}
```

Comment:
Why is this useful?

Reply:
Still relevant.

Reply:
If we remove this, a E0277 error will be emitted because `Box<dyn Trait>` does not implement `Trait`

```
error[E0277]: the trait bound `std::boxed::Box<dyn LateLintPass<'tcx>>: LintPass` is not satisfied
   --> compiler/rustc_lint/src/late.rs:436:45
    |
436 |             let lints = LintPass::get_lints(pass);
    |                         ------------------- ^^^^ the trait `LintPass` is not implemented for `std::boxed::Box<dyn LateLintPass<'tcx>>`
    |                         |
    |                         required by a bound introduced by this call

```

The solution I found at the time was this one.

Reply:
And reborrowing at the call place didn't work?